src/container/include/container/container_rpc.hpp

```
@@ -0,0 +1,400 @@
+              send_heartbeat(socket);
+            }
+          }
+          continue;
+        }
+
+        connected = true;
+        last_activity_time = Clock::now();
+
+        PerformanceTimer::start_timing();
+
+        zmq::message_t msg_delimiter;
+        zmq::message_t msg_msg_type_bytes;
+
+        socket.recv(&msg_delimiter, 0);
+        socket.recv(&msg_msg_type_bytes, 0);
+
+        rpc::MessageType message_type = static_cast<rpc::MessageType>(
+            static_cast<int*>(msg_msg_type_bytes.data())[0]);
+
+        switch (message_type) {
+          case rpc::MessageType::Heartbeat: {
+            log_info(LOGGING_TAG_CONTAINER, "Received heartbeat!");
+            log_event(rpc::RPCEvent::ReceivedHeartbeat);
+            bool requesting_metadata = handle_heartbeat(socket);
+            if (requesting_metadata) {
+              send_container_metadata(model_name, model_version,
+                                      model.get_input_type(), socket);
+            }
+          } break;
+
+          case rpc::MessageType::ContainerContent: {
+            log_event(rpc::RPCEvent::ReceivedContainerContent);
+            zmq::message_t msg_request_id;
+            zmq::message_t msg_request_header;
+
+            socket.recv(&msg_request_id, 0);
+            socket.recv(&msg_request_header, 0);
+
+            int msg_id = static_cast<int*>(msg_request_id.data())[0];
+            RequestType request_type = static_cast<RequestType>(
+                static_cast<int*>(msg_request_header.data())[0]);
+
+            switch (request_type) {
+              case RequestType::PredictRequest:
+                handle_predict_request(model, input_parser, socket,
+                                       input_header_buffer, output_buffer,
+                                       msg_id);
+                break;
+
+              case RequestType::FeedbackRequest:
+                // Do nothing for now
+                break;
+
+              default: break;
```

Probably want to throw an exception or at least log something if we hit the default case

---

Added an exception